.github/workflows/deploy-to-microk8s.yaml

```
@@ -0,0 +1,122 @@
+name: Validation on MicroK8s
+on:
+  workflow_dispatch:
+    inputs:
+      k8s_version:
+        description: 'Kubernetes version to be used for the AKS cluster. Make sure that the corresponding K8s version is supported by the cloud.'
+        required: false
+      uats_branch:
+        description: 'Branch to run the UATs from e.g. main or track/1.10. By default, this is defined by the dependencies.yaml file.'
+        required: false
+  pull_request:
+  schedule:
+    - cron: "17 02 * * 1"
+
+jobs:
+  deploy-solution:
+    name: Deploy CKF + MLFlow solution to K8s
+    runs-on: ["self-hosted-linux-amd64-jammy-xlarge"]
+
+    strategy:
+      fail-fast: false
+      matrix:
+        bundle:
+          - module: kubeflow-mlflow
+            uats:
+              env: uats-remote
+              extra-args: --filter "not feast"
+          - module: kubeflow-feast
+            uats:
+              env: feast-remote
+              extra-args: ""
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v5
+
+      - name: Set envvars from dependencies.yaml
+        run: |
+          yq eval 'to_entries | .[] | "\(.key)=\(.value)"' ".github/dependencies.yaml" | while IFS= read -r line; do
+            echo "$line" >> "$GITHUB_ENV"
+          done
+
+      - name: Update ENV variables from inputs if available
+        run: |
+          K8S_VERSION=${{ inputs.k8s_version || env.K8S_VERSION }}
+          echo "K8S_VERSION=${K8S_VERSION}" >> $GITHUB_ENV
+          UATS_BRANCH=${{ inputs.uats_branch || env.UATS_BRANCH }}
+          echo "UATS_BRANCH=${UATS_BRANCH}" >> $GITHUB_ENV
+
+      - uses: actions/checkout@v4
+
+      - id: setup-python
+        name: Setup Python
+        uses: actions/setup-python@v5.0.0
+        with:
+          python-version: "3.12"
+          architecture: x64
+
+      - name: Install tox & poetry
+        run: |
+          pip install tox
+          pip install poetry
+
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          juju-channel: "3.6/stable"
+          provider: microk8s
+          channel: ${{ format('{0}-strict/stable', env.K8S_VERSION) }}
+          bootstrap-options: "--agent-version 3.6.9"
+          microk8s-group: snap_microk8s
+          microk8s-addons: "rbac dns minio metallb:10.64.140.43-10.64.140.49"
```

suggestion(blocking): I don't think we actually need the `minio` add-on here, which just deploys `minio-operator` as described in https://microk8s.io/docs/addon-minio. When installing via microk8s (which we currently do in our public documentation), we run `sudo microk8s enable dns hostpath-storage metallb:10.64.140.43-10.64.140.49 rbac`
```suggestion
          microk8s-addons: "rbac dns hostpath-storage metallb:10.64.140.43-10.64.140.49"
```

---

Done!